### ISSUE TITLE
Trouble reimplementing the results in "AdapterHub: A Framework for Adapting Transformers" paper

### ISSUE THREAD

**Issue by CSerxy** [Adapter-Hub/adapter-transformers#215]
## Environment info
     
- `adapter-transformers` version: newest version
- Platform:
- Python version: A100
- PyTorch version (GPU?):3.6

## Details
Hi I am trying to reimplementing the results in "AdapterHub: A Framework for Adapting Transformers" paper. However, I cannot get the results in your paper (0.121 F1 for pfeiffer, 0.102 for houlsby). I do get 92.8 for SST-2, by using a similar code. I wonder what is the problem here? Many thanks!

The code is attached below:
`import torch`
`from transformers import BertTokenizer, BertForSequenceClassification`

`import logging`
`logging.basicConfig(level=logging.INFO)`

`tokenizer = BertTokenizer.from_pretrained('bert-base-uncased')`

`model = BertForSequenceClassification.from_pretrained('bert-base-uncased')`

`#adapter_name = model.load_adapter('sts/mrpc@ukp', config='pfeiffer')`
`adapter_name = model.load_adapter('sts/mrpc@ukp', config='houlsby')`

`model.set_active_adapters(adapter_name)`

`ncorrect, nsamples = 0, 0`
`tp, fn, fp, tn = 0.0, 0.0, 0.0, 0.0`
`with open('/glue_data/MRPC/dev.tsv') as fin:`
`    fin.readline()`
`    for index, line in enumerate(fin):`
`        tokens = line.strip().split('\t')`
`        sent1, sent2, target = tokens[3], tokens[4], tokens[0]`
`        input_tensor = torch.tensor([tokenizer.encode(sent1, sent2)])`
`        outputs = model(input_tensor)`
`        prediction_label = str(torch.argmax(outputs[0]).item())`

`        ncorrect += int(prediction_label == target)`
`        nsamples += 1`
`        if target == '1' and prediction_label == '1':`
`            tp += 1`
`        if target == '1' and prediction_label == '0':`
`            fn += 1`
`        if target == '0' and prediction_label == '1':`
`            fp += 1`
`        if target == '0' and prediction_label == '0':`
`            tn += 1`
`print(tp, fn, fp, tn)`
`print('f1 = ' + str(tp / (tp + 0.5 * (fp + fn))))`
`print('| Accuracy: ', float(ncorrect)/float(nsamples))`

**Comment by calpt** [comment]
Hey @CSerxy,

try to modify your input tokenization as follows:
```python
inputs = tokenizer(sent1, sent2, return_tensors="pt")
outputs = model(**inputs)
```
This makes sure the sentence pair is correctly encoded and all required inputs (e.g. also the BERT token type ids) are passed to the model.

**Comment by CSerxy** [comment]
Many thanks for the answer! Will have a try.
